### PR TITLE
add segment tree

### PR DESCRIPTION
**Pull request by ssk4988**
### Submission Link/Screenshots
https://cses.fi/paste/d36090bbadc34191a4b8ca/
image

### Checklist
- [x] AC on at least 1 problem
- [x] Using tabs for indentation
- [x] Longest line at most 63 characters (with tab length 2)
- [x] Well documented

### CODE REVIEW

File: content/data-structures/SegmentTree.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ * Status: tested on https://cses.fi/problemset/task/1649
+ */
+#pragma once
+
+struct Node {
+	Node *l = 0, *r = 0;
+	int lo, hi;
+	ll val = 0;
+	Node(int lo, int hi) : lo(lo), hi(hi) {
+		if (lo + 1 < hi) {
```
Comment (tylerm390) [inline content/data-structures/SegmentTree.h]: return instead of extra indentation?

File: content/data-structures/SegmentTree.h [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	void add(int idx, int v) {
+		if (lo + 1 == hi) {
+			val += v;
+			return;
+		}
+		int mid = lo + (hi - lo) / 2;
+		(mid <= idx ? r : l)->add(idx, v);
+		val = comb(l->val, r->val);
+	}
+	ll comb(ll a, ll b) { return min(a, b); }
```
Comment (tylerm390) [inline content/data-structures/SegmentTree.h]: can we use merge instead of comb?

File: content/data-structures/SegmentTree.h [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ * License: CC0
+ * Source: Inspired by KACTL
+ * Description: Segment tree supporting Range Minimum Queries and Point Add Queries
+ * Time: O(\log N)
+ * Status: tested on https://cses.fi/problemset/task/1649
+ */
+#pragma once
+
+struct Node {
+	Node *l = 0, *r = 0;
```
Comment (tylerm390) [inline content/data-structures/SegmentTree.h]: `l` is annoying to debug. We should use something different

File: content/data-structures/SegmentTree.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+		if (L <= lo && hi <= R) return val;
+		return comb(left->query(L, R), right->query(L, R));
+	}
+	void add(int idx, int v) {
+		if (lo + 1 == hi) {
+			val += v;
+			return;
+		}
+		int mid = lo + (hi - lo) / 2;
+		(mid <= idx ? right : left)->add(idx, v);
```
Comment (azurerazor) [inline content/data-structures/SegmentTree.h]: not necessarily mandating a change here (would like to see what other people think) but this line kinda feels like it obfuscates whats actually happening more than an if statement would.

File: content/data-structures/SegmentTree.h [hunk 5]
```diff
@@ -0,0 +1,38 @@
+/**
+ * Author: Sachin Sivakumar
+ * Date: 2024-10-09
+ * License: CC0
+ * Source: Inspired by KACTL
+ * Description: Segment tree supporting Range Minimum Queries and Point Add Queries
+ * Time: O(\log N)
```
Comment (azurerazor) [inline content/data-structures/SegmentTree.h]: Documentation feels incomplete to me. Flesh out a bit more what a segment tree does and I also think maybe time complexity needs an update as well? Doesn't feel descriptive either

**Review by azurerazor — CHANGES_REQUESTED** [review]

**Review by azurerazor — APPROVED** [review]
lgtm

**Pull request merged**